internal/monitors/cadvisor/kubeletstats.go

```diff
@@ -49,8 +52,21 @@ func (ks *KubeletStatsMonitor) Configure(conf *KubeletStatsConfig) error {
 		return err
 	}
 
+	if ks.Output.HasEnabledMetricInGroup(groupPodEphemeralStats) {
+		conf.IncludePodEphemeralStats = true
+	}
+
 	return ks.Monitor.Configure(&conf.MonitorConfig, ks.Output.SendDatapoint,
-		newKubeletInfoProvider(client))
+		newKubeletInfoProvider(client, conf.IncludePodEphemeralStats))
+}
+
+// GetExtraMetrics returns additional metrics that should be allowed through.
+func (c *KubeletStatsConfig) GetExtraMetrics() []string {
```

Comment:
You can get rid of this function entirely.  For all new stuff we are purely using the metadata and no flags.

Reply:
Removed this method